x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java

```diff
@@ -381,4 +393,61 @@ private void invalidateRefreshToken(RestClient client, String refreshToken) thro
         Response invalidateResponse = client.performRequest(invalidateRequest);
         assertOK(invalidateResponse);
     }
+
+    /**
+     * Hack to account for long-running tests. The max lifetime of a token is 1h, but sometimes our tests take longer so tokens created in
+     * the old cluster may be expired by the time we run tests in the mixed/upgraded clusters.
+     *
+     * This method extends the expiration time of all tokens by writing to the `.security-token` index directly.
+     *
+     * We extend the expiration time for all tokens, instead of selected ones because it requires true hackery to get a hold of a docId
+     * given only an access token and refresh token.
+     */
+    private void extendExpirationTimeForAllTokens() throws Exception {
+        refreshSecurityTokensIndex();
+        final List<String> tokensIds = getAllTokenIds();
+        final var bulkRequest = new Request("POST", "/.security-tokens/_bulk?refresh=true");
+        bulkRequest.setOptions(bulkRequest.getOptions().toBuilder().setWarningsHandler(WarningsHandler.PERMISSIVE));
+        final long newExpirationTime = Instant.now().plus(1, ChronoUnit.HOURS).toEpochMilli();
+        bulkRequest.setJsonEntity(tokensIds.stream().map(tokenId -> """
+            {"update": {"_id": "%s"}}
+            {"doc": {"access_token": {"user_token": {"expiration_time": %s}}}}
+            """.formatted(tokenId, newExpirationTime)).collect(Collectors.joining("\n")));
+        final Response bulkResponse = client().performRequest(bulkRequest);
+        assertOK(bulkResponse);
+        final Map<String, Object> bulkResponseMap = entityAsMap(bulkResponse);
+        assertEquals(false, bulkResponseMap.get("errors"));
+    }
+
+    private void refreshSecurityTokensIndex() throws IOException {
```

Comment:
This may be overkill but I want to avoid weird flakiness here -- it would take forever to debug for someone looking at this in the future.